File: src/com/Torvald/Terrarum/ConsoleCommand/TeleportPlayer.java
package com.Torvald.Terrarum.ConsoleCommand;

import com.Torvald.Terrarum.Game;
import com.Torvald.Terrarum.MapDrawer.MapDrawer;
import com.Torvald.Terrarum.Terrarum;

/**
 * Created by minjaesong on 16-01-24.
 */
public class TeleportPlayer implements ConsoleCommand {

    @Override
    public void execute(String[] args) {
        if (args.length != 3) {
            printUsage();
        }
        else {

            int x, y;
            try {
                x = new Integer((args[1])) * MapDrawer.getTILE_SIZE() + (MapDrawer.getTILE_SIZE() / 2);
                y = new Integer((args[2])) * MapDrawer.getTILE_SIZE() + (MapDrawer.getTILE_SIZE() / 2);
            }
            catch (NumberFormatException e) {
                new Echo().execute("Wrong number input.");
                return;
            }

            Terrarum.game.getPlayer().setPosition(x, y);
        }
    }

    @Override
    public void printUsage() {
        new Echo().execute("Usage: teleport [x-tile] [y-tile]");
    }
}

File: src/net/torvald/terrarum/console/TeleportPlayer.kt
package net.torvald.terrarum.console

import net.torvald.terrarum.StateInGame
import net.torvald.terrarum.mapdrawer.MapDrawer
import net.torvald.terrarum.Terrarum

/**
 * Created by minjaesong on 16-01-24.
 */
internal object TeleportPlayer : ConsoleCommand {

    override fun execute(args: Array<String>) {
        if (args.size != 3) {
            printUsage()
        }
        else {

            val x: Int
            val y: Int
            try {
                x = args[1].toInt() * MapDrawer.TILE_SIZE + MapDrawer.TILE_SIZE / 2
                y = args[2].toInt() * MapDrawer.TILE_SIZE + MapDrawer.TILE_SIZE / 2
            }
            catch (e: NumberFormatException) {
                Echo("Wrong number input.")
                return
            }

            Terrarum.ingame.player.setPosition(x.toDouble(), y.toDouble())
        }
    }

    override fun printUsage() {
        Echo("Usage: teleport [x-tile] [y-tile]")
    }
}
